src/extension.ts

```diff
@@ -197,6 +198,25 @@ export async function activate(context: vscode.ExtensionContext) {
     addDecorations();
   };
 
+  function logBranchTypes(event: vscode.TextEditorSelectionChangeEvent) {
+    const getBranch = (branch: SyntaxNode[]): SyntaxNode[] => {
+      if (branch[0].parent) {
+        return getBranch([branch[0].parent, ...branch]);
+      }
+      return branch;
+    };
+
+    const location = new vscode.Location(
+      vscode.window.activeTextEditor!.document.uri,
+      event.selections[0]
+    );
+    const leaf: SyntaxNode = getNodeAtLocation(location);
+    const branch = getBranch([leaf]);
+    branch.forEach((node, i) => console.log(">".repeat(i + 1), node.type));
+    const leafText = leaf.text.replace(/\s+/g, " ").substring(0, 100);
+    console.log(">".repeat(branch.length), `"${leafText}"`);
+  }
```

Comment:
nice.  yeah this one could be useful.  I think I might use `console.debug` for these tho